test/libsolidity/smtCheckerTests/operators/slice_default_end.sol

```diff
@@ -0,0 +1,17 @@
+pragma experimental SMTChecker;
+
+contract C {
+	function f(bytes calldata b) external pure {
+		require(b.length == 30);
+		require(b[10] == 0xff);
+		require(b[b.length - 1] == 0xaa);
+		assert(bytes(b[10:]).length == 20);
+		assert(bytes(b[10:])[0] == 0xff);
+		assert(bytes(b[10:])[5] == 0xff);
+		assert(bytes(b[10:])[19] == 0xaa);
+	}
+}
+// ----
+// Warning 4661: (221-253): BMC: Assertion violation happens here.
+// Warning 4661: (257-289): BMC: Assertion violation happens here.
+// Warning 4661: (293-326): BMC: Assertion violation happens here.
```

Comment:
I am confused. How come that 3 assertions are failing here?

Reply:
They're false positives. CHC can't solve them because they're too hard, BMC doesn't have array slices so it just reports a false positive.